### PR TITLE
jobs: add image promotion script

### PR DESCRIPTION
**Pull request by winklerm**
Added jobs for promoting built and tested images to production by renaming it according to the following suffixes:

- the production image has `-latest` suffix, 
- the new candidate image has `-latest-new` suffix, 
- the backup of the original production image has `-fallback` suffix.

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Comment by winklerm** [comment]
@mbiarnes FYI

**Comment by winklerm** [comment]
Testing in progress.

**Comment by winklerm** [comment]
Tested on Jenkins.

File: job-dsls/jobs/tools/tools_agent_image_build_PR_test.groovy [hunk 1]
```diff
@@ -56,8 +56,9 @@ job(jobName) {
     parameters {
         stringParam ("OWNER","kiegroup","")
         stringParam ("BRANCH","main","")
-        stringParam ("PACKER_VAR_FILE","packer-kie-rhel7-vars.json","")
-
+        stringParam ("BXMS_JENKINS_BRANCH","master","")
```
Comment (mareknovotny) [inline job-dsls/jobs/tools/tools_agent_image_build_PR_test.groovy:59]: master or main branch?

Comment (winklerm) [reply job-dsls/jobs/tools/tools_agent_image_build_PR_test.groovy:59]: This is the gerrit bxms-jenkins repo still using master.

Comment (mareknovotny) [reply job-dsls/jobs/tools/tools_agent_image_build_PR_test.groovy:59]: ok

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**